### PR TITLE
Add max_depth for prediction

### PR DESCRIPTION
**Pull request by GoodDeeds**

### CODE REVIEW

File: extreme_classification/hierarchicalXC/classifier.py [hunk 1]
```diff
@@ -110,6 +113,14 @@ def traverse_classifiers(self, current_id, this_ids):
                 continue
             classifier_ids = self.merge_iterations[current_id][c]
             if classifier_ids < self.num_classes:
+                assert len(self.merge_indices[classifier_ids]) == 1
```
Comment (vishwakftw) [inline extreme_classification/hierarchicalXC/classifier.py]: Failure message perhaps

File: extreme_classification/hierarchicalXC/classifier.py [hunk 2]
```diff
@@ -110,6 +113,14 @@ def traverse_classifiers(self, current_id, this_ids):
...
+                assert self.merge_indices[classifier_ids][0] == classifier_ids
```
Comment (vishwakftw) [inline extreme_classification/hierarchicalXC/classifier.py]: Here too.

Comment (GoodDeeds) [reply extreme_classification/hierarchicalXC/classifier.py]: Please check

**Comment by vishwakftw** [comment]
Fix Travis and merge conflicts.

**Comment by vishwakftw** [comment]
Lint is complaining.

**Pull request merged**